### PR TITLE
feat(experimentalIdentityAndAuth): enable identity and auth by default

### PR DESCRIPTION
**Pull request by kuhe**
Derived from:

- #1321
- #1322 
- #1323 
- #1324 
- #1325

`experimentalIdentityAndAuth` behavior is now the default auth behavior.
    
The `experimentalIdentityAndAuth` flag is now oppositely replaced with
`useLegacyAuth`, which enables legacy auth behavior for backward
compatibility concerns.

----

Expected diff in clients can be seen in the linked SDK PR.

This PR changes the resolution order of some plugins. This is safe since the plugins declare any dependencies in their input and resolved types such that any hard requirements in ordering will express as TypeScript compilation errors.

This PR moves some private client methods inline.

### CODE REVIEW

**Comment by kuhe** [comment]
taking over from https://github.com/smithy-lang/smithy-typescript/pull/1326, leaving it as reference

**Comment by kuhe** [comment]
The diff for this PR is seen in https://github.com/aws/aws-sdk-js-v3/pull/6299

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/DirectedTypeScriptCodegen.java [hunk 1]
```diff
@@ -102,6 +102,11 @@ public TypeScriptCodegenContext createContext(CreateContextDirective<TypeScriptS
             });
         });
 
+        directive.integrations().forEach(integration -> {
+            LOGGER.info(() -> "Mutating plugins from TypeScriptIntegration: " + integration.getClass().getName());
```
Comment (hpmellema) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/DirectedTypeScriptCodegen.java:106]: nit: Integrations have a `name()` method. I would recommend using that here over the class name.

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java [hunk 2]
```diff
@@ -474,7 +484,29 @@ private void addCommandSpecificPlugins() {
...
+                    List<String> additionalParameters = CodegenUtils.getFunctionParametersList(paramsMap);
+                    Map<String, CommandWriterConsumer> clientAddParamsWriterConsumers =
+                        plugin.getOperationAddParamsWriterConsumers();
+                    if (additionalParameters.isEmpty() && clientAddParamsWriterConsumers.isEmpty()) {
+                        return;
+                    }
+                    writer.writeInline(", { ");
+                    writer.writeInline(String.join(", ", additionalParameters));
+                    clientAddParamsWriterConsumers.forEach((key, consumer) -> {
+                        writer.writeInline(key).writeInline(": ");
```
Comment (hpmellema) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java:497]: consider changing this line to use a consumer parameters. For example: 
```
writer.writerInline("$L: $C, ", key, w -> consumer.accept(w, ...))
```
I find this construction a bit easier to reason about.

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java [hunk 3]
```diff
@@ -474,7 +484,29 @@ private void addCommandSpecificPlugins() {
...
-                writer.write("$pluginFn:T(config" + additionalParamsString + "),");
+                writer.openBlock("$pluginFn:T(config", "),", () -> {
+                    List<String> additionalParameters = CodegenUtils.getFunctionParametersList(paramsMap);
+                    Map<String, CommandWriterConsumer> clientAddParamsWriterConsumers =
+                        plugin.getOperationAddParamsWriterConsumers();
+                    if (additionalParameters.isEmpty() && clientAddParamsWriterConsumers.isEmpty()) {
+                        return;
+                    }
+                    writer.writeInline(", { ");
+                    writer.writeInline(String.join(", ", additionalParameters));
```
Comment (hpmellema) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java:495]: nit: You could use templating and add additional params to the context
```
writer.writeInline(", {${#params}${value:L}${^key.last}, ${/key.last}${/params}")
```

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/ServiceBareBonesClientGenerator.java [hunk 4]
```diff
@@ -453,7 +450,32 @@ private void generateConstructor() {
...
+                            plugin.getClientAddParamsWriterConsumers();
+
+                        if (additionalParameters.isEmpty() && clientAddParamsWriterConsumers.isEmpty()) {
+                            return;
+                        }
+
+                        writer.writeInline(", {");
+                        writer.writeInline(String.join(", ", additionalParameters));
+                        clientAddParamsWriterConsumers.forEach((key, consumer) -> {
+                            writer.writeInline(key).writeInline(": ");
```
Comment (hpmellema) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/ServiceBareBonesClientGenerator.java:465]: see above.

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddEventStreamDependency.java [hunk 5]
```diff
@@ -44,6 +44,13 @@
 @SmithyInternalApi
 public final class AddEventStreamDependency implements TypeScriptIntegration {
 
+    @Override
+    public List<String> runAfter() {
+        return List.of(
+            AddBuiltinPlugins.class.getCanonicalName()
```
Comment (hpmellema) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddEventStreamDependency.java:50]: Use the `name()` method. This is what the Directed codegen filters actually use when determining run order.

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/auth/http/integration/AddHttpAuthSchemePlugin.java [hunk 6]
```diff
@@ -43,45 +41,76 @@
...
     @Override
     public boolean matchesSettings(TypeScriptSettings settings) {
-        return settings.getExperimentalIdentityAndAuth();
+        return !settings.useLegacyAuth();
     }
 
     @Override
     public List<RuntimeClientPlugin> getClientPlugins() {
+        Map<String, ClientWriterConsumer> httpAuthSchemeParametersProvider = Map.of(
+            "httpAuthSchemeParametersProvider", (w, clientBodySection) -> {
```
Comment (hpmellema) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/auth/http/integration/AddHttpAuthSchemePlugin.java:54]: This is really pretty hard to follow. Consider breaking out the consumers into static methods then you can have: 
```
Map.of(
       "httpAuthSchemeParametersProvider", AddHttpAuthSchemePlugin::parametersProvider, 
       ... etc
)
```

**Review by hpmellema — CHANGES_REQUESTED** [review]

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java (hunk 3 above)
Comment (kuhe) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java:495]: I don't understand this syntax at a glance. Are there docs for this?

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddEventStreamDependency.java (hunk 5 above)
Comment (kuhe) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddEventStreamDependency.java:50]: No DI framework. Instantiate one to get the name?

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java (hunk 3 above)
Comment (hpmellema) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java]: https://smithy.io/2.0/guides/building-codegen/generating-code.html

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddEventStreamDependency.java (hunk 5 above)
Comment (hpmellema) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddEventStreamDependency.java:50]: Yeah. That would be my recommendation. Then if the name ever changes you wont have things get out of sync.

**Review by hpmellema — APPROVED** [review]

**Review by trivikr — APPROVED** [review]
